src/mui/input/ReferenceInput.js

```diff
@@ -164,20 +175,24 @@ export class ReferenceInput extends Component {
             onChange,
             children,
             meta,
+            record,
         } = this.props;
-        if (!referenceRecord && !allowEmpty) {
+        if (!matchingReferences.length) {
```

Comment:
not good: for instance, for an autocomplete, if I type a radom string, there will be no matching reference, even once the fetch has succeeded